components/collector/src/source_collectors/azure_devops/inactive_branches.py

```diff
@@ -0,0 +1,62 @@
+"""Azure DevOps Server inactive branches collector."""
+
+from datetime import datetime
+from typing import TypedDict, cast
+
+from base_collectors import BranchType, InactiveBranchesSourceCollector
+from collector_utilities.date_time import parse_datetime
+from collector_utilities.type import URL
+from model import SourceResponses
+
+from .base import AzureDevopsRepositoryBase
+
+
+class Commit(TypedDict):
+    """Commit information as returned by the AzureDevOps API."""
+
+    committer: dict[str, str]
+    url: str
+
+
+class AzureDevOpsBranchType(BranchType):
+    """Azure DevOps branch information as returned by the AzureDevOps API."""
+
+    commit: Commit
+    aheadCount: int  # noqa:  N815
+    isBaseVersion: bool  # noqa:  N815
+
+
+class AzureDevopsInactiveBranches[Branch: AzureDevOpsBranchType](
+    InactiveBranchesSourceCollector, AzureDevopsRepositoryBase
+):
+    """Collector for inactive branches."""
+
+    async def _api_url(self) -> URL:
+        """Extend to add the branches API path."""
+        api_url = str(await super()._api_url())
+        return URL(f"{api_url}/stats/branches?api-version=4.1")
+
+    async def _landing_url(self, responses: SourceResponses) -> URL:
+        """Extend to add the branches path."""
+        landing_url = str(await super()._landing_url(responses))
+        return URL(f"{landing_url}/branches")
+
+    async def _branches(self, responses: SourceResponses) -> list[Branch]:
+        """Return a list of branches from the responses."""
+        return cast(list, (await responses[0].json())["value"])
+
+    def _is_default_branch(self, branch: Branch) -> bool:
+        """Return whether the branch is the default branch."""
+        return branch["isBaseVersion"]
+
+    def _is_branch_merged(self, branch: Branch) -> bool:
+        """Return whether the branch has been merged with the default branch."""
+        return branch["aheadCount"] == 0
```

Comment:
This seems a bit unexpected, but I can't really think of a reason why it would be incorrect. If squash-merges aren't a thing, then I guess it is accurate